charmdir.go

```diff
@@ -29,6 +29,17 @@ type CharmDir struct {
 // Trick to ensure *CharmDir implements the Charm interface.
 var _ Charm = (*CharmDir)(nil)
 
+// IsCharmDir returns true if the path reasonably represents a charm, even
```

Comment:
Standard go phrasing for bool-returning functions uses "reports where",
for example:

    // Is CharmDir reports whether the path is likely to represent
    // a charm, even if it cannot be opened as one.